staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -576,21 +576,21 @@ func TestReflectorFullListIfExpired(t *testing.T) {
 			case "0":
 				return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil
 			case "10":
+				if options.Continue != "" {
+					t.Fatalf("specifying resource version (%s) is not allowed when using continue (%s).", options.ResourceVersion, options.Continue)
+				}
 				switch options.Limit {
 				case 4:
-					switch options.Continue {
-					case "":
-						return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
-					case "C1":
-						return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
-					default:
-						t.Fatalf("Unrecognized Continue: %s", options.Continue)
-					}
-				case 0:
+					return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
+				case 0: // Full re-list
 					return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "11"}, Items: pods[0:8]}, nil
 				default:
 					t.Fatalf("Unrecognized Limit: %d", options.Limit)
 				}
+			case "":
```

Comment:
this is adding a new branch, but not checking limit and continue as exhaustively as the branch it replaced... make sure limit and continue are only the values we expect?

Reply:
Added more checks to make sure all options are as we expect.

Reply:
This test is a bit artificial (we're simulating a particular flow) and hard to understand at the first glance but it's the best thing we have right now.
Adding more checks makes sense to make sure the flow is exactly as it should be but in general we should aim to have a more "e2e/functional" tests here with the real storage. The only problem in that case would be simulating errors. I think this is what you were mentioning in the parent issue of this PR. I will try taking a look at that in my spare cycles.